firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/send/DataTransportCrashlyticsReportSender.java

```diff
@@ -0,0 +1,91 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.crashlytics.internal.send;
+
+import android.content.Context;
+import androidx.annotation.NonNull;
+import com.google.android.datatransport.Encoding;
+import com.google.android.datatransport.Event;
+import com.google.android.datatransport.Transport;
+import com.google.android.datatransport.cct.CCTDestination;
+import com.google.android.datatransport.runtime.TransportRuntime;
+import com.google.firebase.crashlytics.internal.model.CrashlyticsReport;
+import com.google.firebase.crashlytics.internal.model.serialization.CrashlyticsReportJsonTransform;
+import java.nio.charset.Charset;
+import java.util.List;
+
+/**
+ * This class is responsible for sending CrashlyticsReport objects to Crashlytics through Google
+ * DataTransport.
+ */
+public class DataTransportCrashlyticsReportSender {
+
+  private static final CrashlyticsReportJsonTransform TRANSFORM =
+      new CrashlyticsReportJsonTransform();
+  private static final String CRASHLYTICS_ENDPOINT =
+      mergeStrings("hts/cahyiseot-agolai.o/1frlglgc/aclg", "tp:/rsltcrprsp.ogepscmv/ieo/eaybtho");
+  private static final String CRASHLYTICS_API_KEY =
+      mergeStrings("AzSBpY4F0rHiHFdinTvM", "IayrSTFL9eJ69YeSUO2");
+  private static final String CRASHLYTICS_TRANSPORT_NAME = "FIREBASE_CRASHLYTICS_REPORT";
+
+  private final Transport<CrashlyticsReport> transport;
+  private final SendCallback<CrashlyticsReport> sendCallback;
+
+  static DataTransportCrashlyticsReportSender create(
```

Comment:
Is there a good reason to make this new callback type here instead of using a `Task<CrashlyticsReport>`?

It seems like this is the perfect use case for a Promise -- a result or error that gets resolved at most once.

Reply:
I'd originally written this to map to the underlying callback, but I'll take another look. I think you're right.

Reply:
Okay, talked about this in person - going to hold off on this for now, but may move the loop up a level and have a `sendReport` method that returns a single `Task`. Will revisit after all the "thin thread" changes are in.